x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/EnrichStore.java

```diff
@@ -85,81 +138,141 @@ public static void putPolicy(String name, EnrichPolicy policy, ClusterService cl
-                    .metaData(metaData)
-                    .build();
+        // if the index exists, refresh before searching
+        if (enrichIndex != null) {
+            if (enrichIndex.getIndices().isEmpty()) {
+                throw new IllegalStateException("the enrich index has no index metadata");
             }
 
-            @Override
-            public void clusterStateProcessed(String source, ClusterState oldState, ClusterState newState) {
-                handler.accept(null);
-            }
+            final ActionListener<RefreshResponse> refreshAndGetPoliciesListener = ActionListener.wrap(
+                response -> {
+                    if (response.getSuccessfulShards() < enrichIndex.getIndices().get(0).getNumberOfShards()) {
+                        throw illegalState("not all required shards have been refreshed");
+                    }
+                    innerGetPolicies(state, client, listener);
+                },
+                listener::onFailure
+            );
+            client.admin().indices().refresh(new RefreshRequest(ENRICH_INDEX), refreshAndGetPoliciesListener);
+        } else {
+            // return an empty list since the index does not exist
+            listener.onResponse(List.of());
+        }
+    }
 
-            @Override
-            public void onFailure(String source, Exception e) {
-                handler.accept(e);
+    private static void innerGetPolicies(ClusterState state, Client client, ActionListener<Collection<EnrichPolicy.NamedPolicy>> listener) {
+        ensureIndex(state, client, val -> {
+            if (val == null) {
+                ScrollHelper.fetchAllByEntity(client, new SearchRequest(ENRICH_INDEX).scroll(TimeValue.timeValueSeconds(3)), listener,
+                    hit -> {
+                        try {
+                            XContentParser parser = XContentHelper.createParser(NamedXContentRegistry.EMPTY, null, hit.getSourceRef(),
+                                XContentType.JSON);
+                            return new EnrichPolicy.NamedPolicy(hit.getId(), EnrichPolicy.fromXContent(parser));
+                        } catch (IOException e) {
+                            throw new ElasticsearchParseException("Could not parse enrich policy from hit [{}]", hit.getId(), e);
+                        }
+                    }
+                );
+            } else {
+                listener.onFailure(val);
             }
         });
     }
+
+    private static void ensureIndex(ClusterState state, Client client, Consumer<Exception> handler) {
+        final AliasOrIndex enrichIndex = state.metaData().getAliasAndIndexLookup().get(ENRICH_INDEX);
+        if (enrichIndex != null && enrichIndex.isAlias()) {
+            throw new IllegalStateException("The enrich index exists and is an alias");
+        }
+
+        if (enrichIndex == null) {
+            // create the index
+            client.admin().indices().prepareCreate(ENRICH_INDEX)
```

Comment:
Like we discussed in chat, we should either use index template or keep using the create index api call with the right settings and mappings.